gce/burst-training/README.md

```diff
@@ -0,0 +1,242 @@
+the appropriate modifications to that file.
+
+You can run [build-image.sh](./build-image.sh) from your terminal:
+```
+./build-image.sh <INSTANCE-NAME> <IMAGE-FAMILY>
+```
+
+Here `<INSTANCE-NAME>` is a name you would like to assign to the temporary GCE
+instance on which we build the environment as specified in
+[image.sh](./image.sh). This instance will be deleted once the image has built,
+so you won't have to deal with it for very long.
+
+`<IMAGE-FAMILY>` starts a family of images under which Compute Engine stores
+your image. The value of this is that you can update your image over time and,
+by referring to the image family rather than an individual image when you create
+your burst training instances, the most up-to-date image in that family will be
+used in your training jobs.
+
+**Please replace `<INSTANCE-NAME>` and `<IMAGE-FAMILY>` with your own names.**
+
+
+### Training code
+
+[census-analysis.py](./census-analysis.py) contains the training code we will
+run for the Census problem. The only important thing about this code, as far as
+adapting it to your own use case, is the command line interface. This is the
+means by which we will get our burst training instance to pass parameters to the
+training job.
+
+In the case of the Census code, we have to specify three parameters when we run
+the job:
+
+1. `--census-data-path` -- the Cloud Storage path to the training and test data
+
+2. `--model-output-path` -- the Cloud Storage path at which the trainer should
+   store its trained model binary
+
+3. `--cv-iterations` -- the number of hyperparameter configurations that
+   `RandomizedSearchCV` should try
+
+That particular trainer, also allows you to specify a `--mode`, which defaults
+to `train`. This `--mode` argument will be useful to us once the training job
+has completed, as it will allow us to test out the trained model from our local
+environment.
+
+*Note: The [census-analysis.py](./census-analysis.py) trainer uses TensorFlow's
+[tf.gfile](https://www.tensorflow.org/api_docs/python/tf/gfile) to store and
+retrieve data from Cloud Storage. This is the only capacity in which it uses
+TensorFlow. This is also a very useful module to know about -- it allows you to
+interact with Cloud Storage objects through your Python environment just as you
+would interact with files on your own filesystem.*
+
+
+### Execution
+
+The core of the execution step is a call to `python census-analysis.py`.
+However, this call needs to take place on a Compute Engine instance on which the
+image containing our environment has been loaded. Moreover:
+
+1. The `python census-analysis.py` call must be made with the right command line
+   arguments.
+
+2. The Compute Engine instance should be shut down immediately after the `python
+   censys-analysis.py` call has returned. This will keep Compute Engine charges
+   to a minimum.
+
+We use two Compute Engine mechanisms to achieve this behavior:
+
+1. [Instance
```

Comment:
while i read the file to this point, the guide starts to feel more like it's targeting a DevOps person supporting/building their team's own ML infrastructure.  i honestly feel most data scientists (and many machine learning engineers) will find having to learn all these additional concepts (VMs, images, instance metadata, startup script) too much of an overhead even if most steps need to be done only once.

we should hand this to some data scientists and watch them go through the steps.

Reply:
Agreed on needing to experiment with users. Until then, I propose we keep it as it is.

If it feels to DevOps-y, maybe we could extract this into a separate section and flag it as optional, but I would prefer to keep things as flat as possible for now and modify once we get user feedback.